askama_derive/src/generator/filter.rs

```diff
@@ -14,16 +17,16 @@ impl<'a> Generator<'a, '_> {
         &mut self,
         ctx: &Context<'_>,
         buf: &mut Buffer,
-        name: &PathOrIdentifier<'_>,
+        name: &PathOrIdentifier<'a>,
         args: &[WithSpan<'a, Expr<'a>>],
-        generics: &[WithSpan<'a, TyGenerics<'a>>],
         node: Span<'_>,
     ) -> Result<DisplayWrap, CompileError> {
-        let name = match name {
-            PathOrIdentifier::Path(path) => {
-                return self.visit_custom_filter_with_path(ctx, buf, path, args, generics, node);
-            }
-            PathOrIdentifier::Identifier(name) => *name,
+        let (name, generics): (&str, &[WithSpan<'_, TyGenerics<'_>>]) = match name {
```

Comment:
We could cheat here and skip all `ensure_no_generics` calls (which would be nice) but it would require a small refactor of this code, increasing the diff. I suppose it's better as a follow-up?